controllers/registeredservice_controller.go

```diff
@@ -56,7 +65,31 @@ func (r *RegisteredServiceReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	log := log.FromContext(ctx)
 
 	var rs primazaiov1alpha1.RegisteredService
-	if err := r.Client.Get(ctx, req.NamespacedName, &rs); err != nil {
+	err := r.Client.Get(ctx, req.NamespacedName, &rs)
+	if err != nil && errors.IsNotFound(err) {
+		log.Info("Registered Service not found, hanlding delete event")
+		var sc primazaiov1alpha1.ServiceCatalog
+		err = r.Get(ctx, types.NamespacedName{
+			Name:      "primaza-service-catalog",
+			Namespace: req.NamespacedName.Namespace, // "primaza-system",
+		}, &sc)
+		if err != nil {
+			log.Error(err, "Error fetching Service Catalog")
+			return ctrl.Result{}, err
+		}
+
+		si := ServiceInCatalog(sc, req.Name)
+		sc.Spec.Services = append(sc.Spec.Services[:si], sc.Spec.Services[si+1:]...)
+		log.Info("Updating Service Catalog")
+		err = r.Update(ctx, &sc)
+		if err != nil {
```

Comment:
```suggestion
		if err := r.Update(ctx, &sc); err != nil {
```

to stick to [Effective go guidelines](https://go.dev/doc/effective_go#if), we can use this `if`s format when possible. Obviously, that's not mandatory

Reply:
OK